### DISCUSSION TITLE
Using #action to capture a range of information, based on criteria check

### DISCUSSION THREAD

**Discussion by amelsen** [scandum/tintin#127] [Script help]
Alright... oooo... I have a bit of a tricky one I dont know how to solve.

If you look at this screenshot:

![image](https://user-images.githubusercontent.com/3955258/191946025-bee239e5-23cf-4ba5-8fc4-c590ed077e14.png)

Here you will see a range of various ship names. You can see a brief name and number... their ID, their distance, and the coordinates they are placed at.

I'd like to create an action, that 'scan's for a type of ship (match with some sort of predefined list of shipnames), IF one of those ships are within ... say a set range (could be a value of 2000 or something), then it sets a variable $shipfoe, to the ID of that one.

how on EARTH do I set this up?

Thanks.

**Comment by amelsen** [comment]
Here is another example:
![image](https://user-images.githubusercontent.com/3955258/192855501-35de65ff-747c-41f6-a70a-ba92322d62b7.png)

So.. I've used #HIGH to color two ships red. I can't figure out how to make a generic action that captures the values in the line, because the amount of spaces between the numbers varies. 

How to do that?

**Reply by scandum** [reply]
```
#act {%1 (%2) %s %+1..S %s %+1..S %s %+1..S %s %+1..S %s %+1..S} {#line ignore #showme <138>%1 (%2) [%4] [%6] [%8] [%10] [%12]}
```
Can try that. See #help pcre for more info.

**Reply by amelsen** [reply]
OMG! That works... that is _PERFECT_

Only thing I need to understand then Scandum, is:

Is there an ability to create a 'list' of valid ships... Say matching the above.. a list could be:
Victory Destroyer
Imperial Star Destroyer
Mon Calamari Cruiser
etc etc.. 

And then compare %1 with that?

**Comment by amelsen** [comment]
So basically I have created this list:

#VARIABLE {foeships}
{
    {1} {Victory Destroyer}
    {2} {Imperial Star Destroyer}
    {3} {Galactic Voyager}
    {4} {Mon Mothma}
    {5} {Corellian Corvette}
    {6} {Mon Calamari Cruiser}
    {7} {Hydra}
    {8} {Behir}
    {9} {Gorgon}
    {10} {Imperial Gunboat}
    {11} {System Patroller}
    {12} {Rebel Dream}
    {13} {Nebulon-B Frigate}
    {14} {Matalok}
    {15} {Imperial Star Galleon}
}

And what I'd like to do, is:
Check if the before mentioned action stumples upon a match of the above.
If there is a match, then check if the distance is below a certain distance
If it is, then I want to do a specific action.. say update my prompt or something

**Comment by scandum** [comment]
That's best solved using a table.
```
#act {%1 (%2) %s %+1..S %s %+1..S %s %+1..S %s %+1..S %s %+1..S} {#var ships[%1] {{distance}{%4} {x}{%8}{y}{%10}}
```
I'm not sure if I'm assigning %4 %8 and %10 to the right fields, you'd have to fix that if not, but then you can use:
```
#if {$ships[Victory Destroyer][distance] < 1000} {do something}
```

See `#help lists` and `#help variable` for more info.

**Reply by amelsen** [reply]
Yeah I read those already... but you've flipped it around.

What I was hoping for, was to be able to do that actual lookup IN the #action.
So basically in that action, look if the shiptype IS one of the ones in the list, and IF it is, then do something, if distance is less than a certain amount... ?
With your #if statement, if I understand it correctly.. then I'd have to do 14 if statements to check for each ship type? Or?

**Reply by amelsen** [reply]
Just to clarify... the above predefined list is the actual list of ships that has a relevance and that the action has to match up against... 

what I want to accomplish in the #action, is to figure out if it stumples upon once of those ships, by matching the shiptype with the list.. if the action sees a match with the shiptypes listed in the predefined $foeships I showed above.. THEN it has to do something?

Dunno if that makes more sense?

**Reply by scandum** [reply]
Easiest is a table for that as well:
```
#VARIABLE {foeships}
{
{Victory Destroyer} {}
{Imperial Star Destroyer} {}
{Galactic Voyager} {}
etc
}

```
#if {&foeships[%1]} {do something}
```

**Reply by amelsen** [reply]
sweet... that works... thanks so much.
